src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -442,39 +442,72 @@ impl AstDisplay for CsvColumns {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum SourceIncludeMetadataType {
-    Key,
-    Timestamp,
-    Partition,
-    Offset,
-    Headers,
-}
-
-impl AstDisplay for SourceIncludeMetadataType {
-    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
-        match self {
-            SourceIncludeMetadataType::Key => f.write_str("KEY"),
-            SourceIncludeMetadataType::Timestamp => f.write_str("TIMESTAMP"),
-            SourceIncludeMetadataType::Partition => f.write_str("PARTITION"),
-            SourceIncludeMetadataType::Offset => f.write_str("OFFSET"),
-            SourceIncludeMetadataType::Headers => f.write_str("HEADERS"),
-        }
-    }
-}
-impl_display!(SourceIncludeMetadataType);
-
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct SourceIncludeMetadata {
-    pub ty: SourceIncludeMetadataType,
-    pub alias: Option<Ident>,
+pub enum SourceIncludeMetadata {
+    Key {
+        alias: Option<Ident>,
+    },
+    Timestamp {
+        alias: Option<Ident>,
+    },
+    Partition {
+        alias: Option<Ident>,
+    },
+    Offset {
+        alias: Option<Ident>,
+    },
+    Headers {
+        alias: Option<Ident>,
+    },
+    Header {
+        alias: Ident,
+        key: String,
```

Comment:
```suggestion
        key: String,
        alias: Ident,
```

Nit: it's nice to have the order of fields in the AST match how they're parsed.